DALOptimizer/MatchExpr.cs

```diff
@@ -0,0 +1,265 @@
+                    script.RemoveText(start, end - start);
+                }
+                if (Pat.SqlDtAdptStmt().Match(expr).Success)
+                {
+                    string MthdretType = expr.GetParent<MethodDeclaration>().ReturnType.GetText();
+                    if (MthdretType == "DataTable" || MthdretType == "System.Data.DataTable")
+                    {
+                        string varName = expr.GetText().Split("()".ToCharArray())[1];
+                        var getDtTbl = Pat.GetDtTbl(varName);
+                        script.Replace(expr, getDtTbl);
+                    }
+                    else if (MthdretType == "DataSet" || MthdretType == "System.Data.DataSet")
+                    {
+                        string varName = expr.GetText().Split("()".ToCharArray())[1];
+                        var getDtSet = Pat.GetDtSet(varName);
+                        script.Replace(expr, getDtSet);
+                    }
+                    else
+                    {
+                        //set method return type as dummy
+                        int Len = expr.GetParent<MethodDeclaration>().ReturnType.ToString().Length;
+                        int startOffset = script.GetCurrentOffset(expr.GetParent<MethodDeclaration>().ReturnType.StartLocation);
+                        script.RemoveText(startOffset, Len);
+                        script.InsertText(startOffset, "DummyText");
+                    }
+                }
+                if (Pat.varDeclMthd().Match(expr).Success)
+                {
+                    if (expr.Parent.GetType().Name == "BlockStatement" &&
+                        expr.Parent.Parent.GetType().Name == "MethodDeclaration")
+                    {
+                        var varName = expr.FirstChild.NextSibling.FirstChild.GetText();
+                        script.Replace(expr, Pat.varDeclMthd1(varName));
+                    }
+                }
+            }
+
+        }
+        private void ExprStatement(DocumentScript script, CSharpFile file)
+        {
+            foreach (var expr in file.IndexOfExprStmt)
+            {
+                var copy = (ExpressionStatement)expr.Clone();
+
+                AllPatterns Pat = new AllPatterns();
+                if (Pat.FillExpr().Match(expr).Success) { script.Remove(expr, true); continue; }
+                if (Pat.StoredProc().Match(expr).Success) { script.Remove(expr, true); continue; }
+                if (Pat.sqlConnstmt().Match(expr).Success) { script.Remove(expr, true); continue; }
+                if (Pat.ConnOpenExprStmt().Match(expr).Success) { script.Remove(expr, true); continue; }
+                if (Pat.ConnCloseExprStmt().Match(expr).Success) { script.Remove(expr, true); continue; }
+                if (Pat.CmdDisposeExprStmt().Match(expr).Success) { script.Remove(expr, true); continue; }
+                if (Pat.ConvertToInt32().Match(expr).Success) { script.Remove(expr, true); continue; }
```

Comment:
The block `{ script.Remove(expr, true); continue; }` is same for all `if` conditions. Can be done in a single line.